File: demos/node/pdf.js
import {
  GoogleGenAI,
  createUserContent,
  createPartFromUri,
} from "@google/genai";

const MODEL_NAME = "gemini-2.0-flash";
const API_KEY = process.env.GEMINI_API_KEY;

const ai = new GoogleGenAI({ apiKey: API_KEY });

async function processDoc(img) {

  const image = await ai.files.upload({
	file: img,
  });

  const result = await ai.models.generateContent({
	model:MODEL_NAME,
	contents: 
	  createUserContent([
		'Write a one paragraph summary of this document, followed by three to four bullet points', 
		createPartFromUri(image.uri, image.mimeType)
	  ])
  })
  
  return result.text;
}

if(process.argv.length < 3) {
  console.log('Pass a path to a PDF.');
  process.exit();
}

(async () => {

  let pdf = process.argv[2];
  console.log(`Asking Gemini about: ${pdf}`);
  let result = await processDoc(pdf);
  console.log(result);

})();

File: demos/python/pdf.py
import sys
import os
from google import genai

client = genai.Client(api_key=os.environ["GEMINI_API_KEY"])


def processDoc(path):
	file_ref = client.files.upload(file=path)
	prompt = 'Write a one paragraph summary of this document, followed by three to four bullet points'
	response = client.models.generate_content(
		model="gemini-2.0-flash", contents=[prompt, file_ref]
	)
	return response.text

if len(sys.argv) == 1:
	print("Pass a path to the PDF.")
	sys.exit()

file = sys.argv[1]

results = processDoc(file)
print(results)
